docs/terminals.md

```diff
@@ -0,0 +1,141 @@
+# Terminals and Standard IO
+All processes on Unix (and Unix-like) operating systems have 3 standard file descriptors (fd) passed to them at start, collectively standard-IO (`stdio`):
+
+* `0`: standard-in (`stdin`), the input stream into the process
+* `1`: standard-out (`stdout`), the output stream from the process
+* `2`: standard-error (`stderr`), the error stream from the process
+
+When creating and running a container via `runc`, it is important to take care to structure the stdio the new container's process receives.
+
+## Terminal Modes
+
+`runc` supports two distinct methods for passing stdio to the container's primary process:
+
+* pass-through
+* new terminal
+
+### Pass-Through
+In pass-through mode, the `stdio` passed to the `runc` call will be passed as is to the primary container. This means that if you do the following:
+
+```
+$ echo input | runc run some_container > /tmp/log.out 2>& /tmp/log.err
+```
+
+Then the stdout of `some_container` will appear in the file `/tmp/log.out`, the stderr will appear in `/tmp/log.err`, and the process will receive input of `input`. `runc`'s stdout was redirected to `/tmp/log.out`, and `runc`, in turn, passed that handle to the container itself.
+
+To invoke pass-through mode, configure your container's `config.json` with `terminal: false` (which is the default).
```

Comment:
It's not the default if you use `runc spec`.

Reply:
Well, it still is the default if no `terminal:` setting is specific. `runc spec` just happens to actually specify it in the `config.json`. But I don't mind updating it.